### PR TITLE
Add metrics endpoint scrape job data source resource

### PR DESCRIPTION
**Pull request by fridgepoet**
This PR contributes to work on Metrics Endpoint scrape jobs.
* Adds the data source resource for metrics endpoint scrape jobs
* Registers data source resources and managed resources with the provider

Closes https://github.com/grafana/cloud-onboarding/issues/7769

Special notes:
For the acceptance test, I followed https://discuss.hashicorp.com/t/acceptance-testing-test-a-data-source/44940 for some guidance.

### CODE REVIEW

File: internal/resources/connections/resource_metrics_endpoint_scrape_job.go [hunk 1]
```diff
@@ -28,8 +28,6 @@ type resourceMetricsEndpointScrapeJob struct {
 	client *connectionsapi.Client
 }
 
-var Resources = makeResourceMetricsEndpointScrapeJob()
```
Comment (fridgepoet) [inline internal/resources/connections/resource_metrics_endpoint_scrape_job.go:30]: These were moved to resources.go

File: internal/resources/connections/resources.go [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+		return nil, fmt.Errorf("unexpected DataSource Configure Type: %T, expected *common.Client", req.ProviderData)
+	}
+
+	if client.ConnectionsAPIClient == nil {
+		resp.Diagnostics.AddError(
+			"The Grafana Provider is missing a configuration for the Cloud Provider API.",
+			"Please ensure that cloud_provider_url and cloud_provider_access_token are set in the provider configuration.",
+		)
+
+		return nil, fmt.Errorf("CloudProviderAPI is nil")
```
Comment (matthewnolf) [inline internal/resources/connections/resources.go]:
```suggestion
		return nil, fmt.Errorf("ConnectionsAPI is nil")
```

File: internal/resources/connections/resources.go [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+			"Unexpected DataSource Configure Type",
+			fmt.Sprintf("Expected *common.Client, got: %T. Please report this issue to the provider developers.", req.ProviderData),
+		)
+
+		return nil, fmt.Errorf("unexpected DataSource Configure Type: %T, expected *common.Client", req.ProviderData)
+	}
+
+	if client.ConnectionsAPIClient == nil {
+		resp.Diagnostics.AddError(
+			"The Grafana Provider is missing a configuration for the Cloud Provider API.",
```
Comment (matthewnolf) [inline internal/resources/connections/resources.go]:
```suggestion
			"The Grafana Provider is missing a configuration for the Connections API.",
```

File: internal/resources/connections/resources.go [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+			"Please ensure that cloud_provider_url and cloud_provider_access_token are set in the provider configuration.",
```
Comment (matthewnolf) [inline internal/resources/connections/resources.go]:
```suggestion
			"Please ensure that connections_api_url and connections_access_token are set in the provider configuration.",
```

**Review by matthewnolf — APPROVED** [review]

**Pull request merged**